test: autodetect available browsers in karma

lundibundi opened the pull request:
- [x] use karma-detect-browsers to autodetect available browsers
- [x] add edge and opera browsers to karma configuration

I haven't added [karma-ie-launcher](https://github.com/karma-runner/karma-ie-launcher)  as it has been last updated quite a while ago and has a lot of issues, so I'm not sure if we should add it.

Fixes: https://github.com/metarhia/jstp/issues/155

aqrln commented on .travis.yml:
```diff
@@ -32,6 +32,7 @@ addons:
       - g++-6
   firefox: latest
 before_script:
+  - export CHROME_BIN=chromium-browser
```
I don't thing it will work because of sandboxing issues, but let's see what Travis says. Maybe it will work this time.

lundibundi replied:
@aqrln yeah, I wasn't sure either but it was worth a try. It failed. Any ideas?

aqrln commented on package.json:
```diff
@@ -46,11 +46,14 @@
     "eslint": "^3.12.2",
     "eslint-plugin-import": "^2.2.0",
     "karma": "^1.3.0",
+    "karma-mocha": "^1.3.0",
```
The order of packages is messed up here. Did you edit it manually?

lundibundi replied:
@aqrln yeah, completely forgot about npm commands =). Got used to editing config files manually that much

aqrln commented:
I suppose it should also fix https://github.com/metarhia/jstp/issues/149.

lundibundi commented:
@aqrln I think so too but I haven't tested it therefore I didn't add it. I'll probably test it later before landing it and add fixes if it works.

aqrln commented:
@lundibundi Windows fails with `Cannot load browser "IE": it is not registered`.
`npm i -D karma-ie-launcher` fixes this. The tests themselves fail with `SyntaxError`s on IE and Edge though, but that's a matter of a different PR.

aqrln commented:
Oh, and can you please rebase onto master and update `package-lock.json`?

lundibundi commented:
@aqrln thanks for testing it on windows. No problem.

lundibundi commented:
@aqrln I've updated package.json and added fixes 149, as In my opinion the issue was about starting karma tests on windows and we have to create separate issue dedicated to fixing the errors detected.

Also Arch Linux still hasn't updated npm version, could you please land it and update `package-lock.json` upon landing?

aqrln commented:
@lundibundi 

> Also Arch Linux still hasn't updated npm version, could you please land it and update package-lock.json upon landing?

`npm i -g npm@5` ;)
But np, I can update it myself.

lundibundi commented:
@aqrln I obviously know of that, but it'll probably break pacman packet and I don't really want to fix with that. :smile:

aqrln commented:
@lundibundi that won't break the pacman package. npm bundled with Node.js and npm installed as a regular global package reside in different directories.

aqrln commented:
Landed in 6339bb0ce01847fd90f53d94e4581e0d78a2226b.
